### DISCUSSION TITLE
文件路径状态问题

### DISCUSSION THREAD

**Discussion by JAINKRE** [1Panel-dev/1Panel#5162] [Q&A]
### 1Panel 版本

v1.10.9-lts

### 问题描述

手动输入文件路径，刷新页面后又恢复原路径，如下：
![20240527-123351](https://github.com/1Panel-dev/1Panel/assets/63288007/0e6c927b-ff52-4603-bcd6-145720f1f302)

### 重现步骤

结果排查发现文件路径现在的实现是根据解析route路由，当修改route路径时可以正常改变文件路径：
![20240527-124914](https://github.com/1Panel-dev/1Panel/assets/63288007/915e4d86-5e4b-4c6a-987a-8f6cf74a3963)

可以考虑通过当用户手动改变文件路径时，通过localStorage.setItem记录，当页面重新刷新时通过localStorage.getItem和路由query比对，当不一致时优先转跳localStorage存储的值。

当然，这只是建议，如果有更好的实现逻辑更好！

**Comment by zhengkunwang223** [comment]
感谢建议 
当前我们已经存储了一份在 localStorage 中，如果你是手动点击文件菜单，会根据 localStorage 去访问，但是如果是从其他地方快速跳转过来，比如网站，那么会根据路由的参数来确定
想象一下：你 localStorage 存了 /opt/1panel  但是你点击网站那网站目录跳转的时候，如果优先读取了 localStorage ，那你就跳转不到网站目录了
